WellLog.Lib/Models/DLIS/LogicalRecordSegmentHeader.cs

```diff
@@ -1,79 +1,68 @@
+            set { LogicalRecordSegmentAttributes = LogicalRecordSegmentAttributes.SetBit(PREDECESSOR_MASK, value); }
         }
 
         public bool Successor
         {
-            get { return GetFlag(LogicalRecordSegmentAttributes, SUCCESSOR_FLAG); }
-            set { LogicalRecordSegmentAttributes = SetFlag(LogicalRecordSegmentAttributes, SUCCESSOR_FLAG, value); }
+            get { return LogicalRecordSegmentAttributes.GetBit(SUCCESSOR_MASK); }
+            set { LogicalRecordSegmentAttributes = LogicalRecordSegmentAttributes.SetBit(SUCCESSOR_MASK, value); }
         }
 
         public bool Encryption
         {
-            get { return GetFlag(LogicalRecordSegmentAttributes, ENCRYPTION_FLAG); }
-            set { LogicalRecordSegmentAttributes = SetFlag(LogicalRecordSegmentAttributes, ENCRYPTION_FLAG, value); }
+            get { return LogicalRecordSegmentAttributes.GetBit(ENCRYPTION_MASK); }
+            set { LogicalRecordSegmentAttributes = LogicalRecordSegmentAttributes.SetBit(ENCRYPTION_MASK, value); }
         }
 
         public bool EncryptionPacket
         {
-            get { return GetFlag(LogicalRecordSegmentAttributes, ENCRYPTION_PACKET_FLAG); }
-            set { LogicalRecordSegmentAttributes = SetFlag(LogicalRecordSegmentAttributes, ENCRYPTION_PACKET_FLAG, value); }
+            get { return LogicalRecordSegmentAttributes.GetBit(ENCRYPTION_PACKET_MASK); }
+            set { LogicalRecordSegmentAttributes = LogicalRecordSegmentAttributes.SetBit(ENCRYPTION_PACKET_MASK, value); }
         }
 
         public bool Checksum
         {
-            get { return GetFlag(LogicalRecordSegmentAttributes, CHECKSUM_FLAG); }
-            set { LogicalRecordSegmentAttributes = SetFlag(LogicalRecordSegmentAttributes, CHECKSUM_FLAG, value); }
+            get { return LogicalRecordSegmentAttributes.GetBit(CHECKSUM_MASK); }
+            set { LogicalRecordSegmentAttributes = LogicalRecordSegmentAttributes.SetBit(CHECKSUM_MASK, value); }
         }
 
         public bool TrailingLength
         {
-            get { return GetFlag(LogicalRecordSegmentAttributes, TRAILING_LENGTH_FLAG); }
-            set { LogicalRecordSegmentAttributes = SetFlag(LogicalRecordSegmentAttributes, TRAILING_LENGTH_FLAG, value); }
+            get { return LogicalRecordSegmentAttributes.GetBit(TRAILING_LENGTH_MASK); }
+            set { LogicalRecordSegmentAttributes = LogicalRecordSegmentAttributes.SetBit(TRAILING_LENGTH_MASK, value); }
         }
 
         public bool Padding
         {
-            get { return GetFlag(LogicalRecordSegmentAttributes, PADDING_FLAG); }
-            set { LogicalRecordSegmentAttributes = SetFlag(LogicalRecordSegmentAttributes, PADDING_FLAG, value); }
+            get { return LogicalRecordSegmentAttributes.GetBit(PADDING_MASK); }
+            set { LogicalRecordSegmentAttributes = LogicalRecordSegmentAttributes.SetBit(PADDING_MASK, value); }
         }
     }
 }
```

Comment:
Is an instance of this class unusable without any of these properties? If so they should be readonly and put in the constructor instead of property sets.

Why?
If you can justify having an "IsValid" method on an object because you need to check if all of the values have been set then you should move those to the constructor so the object can't exist in an invalid state.

Reply:
SetBit does not require a mask.  There are two approaches- pass the "position" or pass the value.  In this case, position 2 is the same as value 4 (2^position).  Simply or (|) the value with the bit you want to set.  For instance,
SetBit(int position = 0)
{
    value |=(2 ^ position);
}
GetBit(int position = 0)
{
    return value &= ~(2 ^ position);
}

Reply:
@RyanDawkins , validation logic is not taking place in this class.  the properties simply encapsulate the logic of using bits within a byte as boolean flags.

@TomMarks , using bitwise operations, such as shifting, OR'ing, and AND'ing is faster that using integer arithmetic, especially multiplication, to achieve the same results.

Reply:
Forgot ResetBit:
ResetBit(int position = 0)
{
value &= ~(2 ^ position);
}